File: api/presets.py
"""Resize presets. web/lib/presets.ts mirrors this list; change both together."""

from dataclasses import dataclass


@dataclass(frozen=True)
class Preset:
    label: str
    width: int
    height: int
    dpi: int | None = None  # written into the file; only print sizes need it


def _mm_to_px(mm: float, dpi: int) -> int:
    return round(mm / 25.4 * dpi)


PRESETS: dict[str, Preset] = {
    "instagram-post": Preset("Instagram post", 1080, 1080),
    "instagram-story": Preset("Instagram story", 1080, 1920),
    "x-header": Preset("X header", 1500, 500),
    "linkedin-banner": Preset("LinkedIn banner", 1584, 396),
    "whatsapp-dp": Preset("WhatsApp DP", 640, 640),
    "passport": Preset("Passport 35x45 mm", _mm_to_px(35, 300), _mm_to_px(45, 300), dpi=300),
}

File: web/lib/presets.ts
/** The resize presets. A mirror of api/presets.py; presets.test.ts checks they agree. */
export const PRESETS = [
  { id: "instagram-post", label: "Instagram post", width: 1080, height: 1080 },
  { id: "instagram-story", label: "Instagram story", width: 1080, height: 1920 },
  { id: "x-header", label: "X header", width: 1500, height: 500 },
  { id: "linkedin-banner", label: "LinkedIn banner", width: 1584, height: 396 },
  { id: "whatsapp-dp", label: "WhatsApp DP", width: 640, height: 640 },
  { id: "passport", label: "Passport 35x45 mm", width: 413, height: 531, dpi: 300 },
] as const;

export type PresetId = (typeof PRESETS)[number]["id"];
